Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -279,17 +279,22 @@ static void softlinkDataDetectorsFrameworks()
 
 #if HAVE(VIDEO_RESTRICTED_DECODING)
 #if PLATFORM(MAC)
+    OSObjectPtr<dispatch_semaphore_t> codeCheckSemaphore;
     if (SandboxExtension::consumePermanently(parameters.trustdExtensionHandle)) {
         // Open up a Mach connection to trustd by doing a code check validation on the main bundle.
         // This is required since launchd will be blocked after process launch, which prevents new Mach connections to be created.
         // FIXME: remove this once <rdar://90127163> is fixed.
-        auto bundleURL = adoptCF(CFBundleCopyBundleURL(CFBundleGetMainBundle()));
-        SecStaticCodeRef code = nullptr;
-        SecStaticCodeCreateWithPath(bundleURL.get(), kSecCSDefaultFlags, &code);
-        if (code) {
-            SecStaticCodeCheckValidity(code, kSecCSDoNotValidateResources, nullptr);
-            CFRelease(code);
-        }
+        codeCheckSemaphore = adoptOSObject(dispatch_semaphore_create(0));
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), [codeCheckSemaphore = codeCheckSemaphore] {
```

Comment:
Probably worth comment here that this dispatch is a perf workaround to skip some main thread only work.

Reply:
Doesn't this cause the main thread to wait on a background thread which presumably has lower priority?

Reply:
Yes, the main thread waits. It's a weird thing to do. The reason we do this is that the underlying framework has some arguably arguably ill-advised logging that it performs if and only if it's called on the main thread. So calling it on a secondary thread is a workaround.

That said, probably DISPATCH_QUEUE_PRIORITY_DEFAULT is a similar priority?

Reply:
> Probably worth comment here that this dispatch is a perf workaround to skip some main thread only work.

Ah, good point! I will add a comment.

Reply:
I think we should use higher priority of the thread. If the work hadn't finished, we wouldn't want the main thread to hang on a lower priority background thread.